### PR TITLE
Addressing Discriminatory Language in Technology

### LINKED ISSUE #28622: Addressing Discriminatory Language in Technology
This issue is automatically created based on existing pull request: magento/magento2#28615: Addressing Discriminatory Language in Technology

---------
### Description (*)
**Blacklist** and **Whitelist** has negative and abusive correlations and is considered as racism. Thus, we should immediately replace these with `allow` and `deny` list.

Due to some technical issues, not all occurrences can be replaced yet, but this is a good start to make a world better.

### Actual result (*)

### PR DESCRIPTION
**Pull request by lbajsarowicz**
### Description (*)
**Blacklist** and **Whitelist** has negative and abusive correlations and is considered as racism. Thus, we should immediately replace these with `allow` and `deny` list.

Due to some technical issues, not all occurrences can be replaced yet, but this is a good start to make a world better.

### Related Pull Requests
https://github.com/magento/adobe-stock-integration/pull/1421

### Fixed Issues (if relevant)
N/A

### Manual testing scenarios (*)
N/A

### Questions or comments
N/A

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#28622: Remove abusive and racist elements from Magento

### CODE REVIEW

**Comment by adeptofvoltron** [comment]
@lbajsarowicz using the opportunity please consider also removing words like *master* and *slave*
image

They also may have abusive correlation

**Comment by lbajsarowicz** [comment]
@magento run all tests

**Comment by PascalBrouwers** [comment]
Thanks for putting in the effort for this @lbajsarowicz !

**Comment by JeroenVanLeusden** [comment]
Maybe “blocklist” and “allowlist” would be better?

**Comment by lenaorobei** [comment]
@magento run all tests

**Comment by lenaorobei** [comment]
@lbajsarowicz tests are failing because of `PHP Fatal error:  Uncaught Error: Cannot instantiate interface Magento\MediaGalleryApi\Model\ExcludedPatternsConfigInterface in /var/www/html/lib/internal/Magento/Framework/ObjectManager/Factory/Dynamic/Developer.php:50`

**Comment by okorshenko** [comment]
@magento create issue

**Comment by lbajsarowicz** [comment]
@magento run all tests

@lenaorobei I fixed the missing `<preference>` hope that's all :)

**Comment by okorshenko** [comment]
@lenaorobei @lbajsarowicz Hi! And thank you for working on that. Are you going to apply the same fix to other code repos? 
@lenaorobei these are breaking changes, we might need to initiate approval request process for 2.4.0 release

**Comment by lenaorobei** [comment]
@magento run all tests

**Comment by lenaorobei** [comment]
Added related PR.

@magento run all tests

File: app/code/Magento/MediaGalleryApi/Api/IsPathBlacklistedInterface.php [hunk 1]
```diff
@@ -8,9 +8,11 @@
...
- * Check if the path is blacklisted for media gallery.
+ * Check if the path is excluded for media gallery.
  *
- * Directory path may be blacklisted if it's reserved by the system.
+ * Directory path may be excluded if it's reserved by the system.
+ * @deprecated Replaced due to abusive word usage
+ * @see \Magento\MediaGalleryApi\Api\IsPathExcludedInterface
  * @api
  */
 interface IsPathBlacklistedInterface
```
Comment (paliarush) [inline app/code/Magento/MediaGalleryApi/Api/IsPathBlacklistedInterface.php]: @lenaorobei  Renaming `@api` interfaces is a MAJOR change.

Comment (lenaorobei) [reply app/code/Magento/MediaGalleryApi/Api/IsPathBlacklistedInterface.php]: This interface was introduced in `2.4.x` release line and is not released yet.

**Comment by lenaorobei** [comment]
@magento run all tests

**Comment by okorshenko** [comment]
i’ve also heard “allow” and “deny” terminologies being suggested as the word to use

**Comment by lbajsarowicz** [comment]
> i’ve also heard “allow” and “deny” terminologies being suggested as the word to use

That is a good point, actually as a replacement to `include`, use `allow` :)

**Comment by lbajsarowicz** [comment]
@magento run all tests

**Comment by fascinosum** [comment]
@lbajsarowicz [here is](https://github.com/rails/rails/issues/33677#issue-352750519) a good explanation of why **allowlist** and **denylist** is better. It is harder to say that something is **_excludelisted_**

**Comment by lbajsarowicz** [comment]
@magento run all tests
@fascinosum You are right, adjusted the changes.

**Comment by lbajsarowicz** [comment]
@magento run all tests

**Comment by akaplya** [comment]
@lenaorobei  @lbajsarowicz, please rename `main` to `leader` and `replica`, main could be uncertain, and multi-leader sounds more convenient than multi-main

**Comment by blacklistisnotracist** [comment]
I do agree the newly suggested terms may be better because they are more descriptive. That being said, calling blacklist or whitelist a racist word is simply wrong.

First, blacklist's origins have nothing to do with black people. What's more, it has never been used as an abusive word against black people. It's only a certain group of extreme non-black liberals who find the word offensive and try to make them look like it has racist connotations. It's as ridiculous as people who think wearing Chinese clothes is offensive for Chinese people unless they are worn by Chinese people.

Black, as a color, carries negative connotations across many cultures. It's so embedded that hunting down every word from every language is almost impossible and meaningless.

Let's stop calling a group of people black first. Let's start from there. Color comes first, race came later. Colors are tied to images, which can't be changed.

Let's stop the hypocrisy and focus on something that have real impacts.

**Review by choinek — COMMENTED** [review]
Dude I hope that's some kind of weird joke. "Black list" != "black human". Your pull request is racist. No one ever connected black list with human races.

**Comment by lbajsarowicz** [comment]
@Choinek Please watch the video from MageTitans Manchester 2018. **There are** people who consider these words abusive even if you personally do not have such correlations.

https://youtu.be/B5gZiOtgJMY?t=341

**Comment by naitsirch** [comment]
@AGPDev Yes.

I agree with others about master/slave being abusive. Change it. But claiming blacklist being racist, just because of "black" is nonsense.

**Comment by lbajsarowicz** [comment]
@magento run all tests

**Comment by lbajsarowicz** [comment]
@magento run all tests

Fixed XSD issues

**Comment by lbajsarowicz** [comment]
@magento run all tests

Fixed leftover in XML file.

**Comment by lbajsarowicz** [comment]
@magento run all tests

**Comment by lbajsarowicz** [comment]
@magento run all tests

**Comment by lbajsarowicz** [comment]
@magento run all tests

There were issues in the code already, that I haven't noticed before. With introducing changes to these files, this turned out that I need to fix them.

**Comment by lbajsarowicz** [comment]
@magento run all tests
Final set of changes, now all checks should pass (only Static Checks failed last time).

**Comment by lbajsarowicz** [comment]
@magento run all tests

**Comment by lbajsarowicz** [comment]
@magento run all tests

**Comment by PascalBrouwers** [comment]
Still hoping this gets merged so it's more descriptive ❤️

**Review by lenaorobei — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-7687](https://jira.corp.magento.com/browse/ENGCOM-7687) has been created to process this Pull Request
:eight_spoked_asterisk: @lenaorobei, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Pull request merged**